sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs

```diff
@@ -344,16 +344,22 @@ private static string GetCanonicalName(string account, string fileSystemName, st
         /// </summary>
         private void EnsureState()
         {
-            if (ExpiresOn == default)
+            if (string.IsNullOrEmpty(Identifier))
             {
-                throw Errors.SasMissingData(nameof(ExpiresOn));
-            }
-            if (string.IsNullOrEmpty(Permissions))
-            {
-                throw Errors.SasMissingData(nameof(Permissions));
+                if (string.IsNullOrEmpty(Permissions))
```

Comment:
If Permissions/ExpiresOn needs to be omitted when there is an access policy, should we also throw if Permissions/ExpiresOn are not null in the case of Identifier being set?

Reply:
For Permissions, yes.  ExpiresOn is allowed if the Access Policy associated with the Identifier doesn't have the ExpiresOn property set.

Reply:
It is worth asking if we should be doing that validation (or even the existing validation) though as this seems like service parameter validation which is something we are not supposed to do.

Reply:
I think we should, because the service doesn't give any indication of what the user did wrong, just a 403.

Reply:
The [guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation) actually cover this and mention that we should work with the service team in that case to improve the error message. But I guess that may not be realistic to do on our timelines. Can we submit an issue for the service team to address this in some way and we can have a corresponding issue to remove the validation if they ever do?

Reply:
It looks like this check was removed in a recent commit. I thought we wanted to add this extra validation?

Reply:
Based on comment from @gapra-msft we shouldn't be adding this as the rule for omitting the Permissions should only applies if Permissions is specified in stored access policy, but we have no way of checking this.